app/controllers/finders_controller.rb

```diff
@@ -196,4 +198,143 @@ def include_ecommerce?
       /government/organisations/hm-revenue-customs/contact
     ].exclude?(content_item.base_path)
   end
+
+  def identify_and_redirect_legacy_params
+    redirect_legacy_params if legacy_params_present?
+  end
+
+  def legacy_params_present?
+    legacy_params = %i[
+      departments
+      from_date
+      official_document_status
+      publication_filter_option
+      publication_type
+      subtaxons
+      taxons
+      to_date
+    ].freeze
+    given_params = params.keys.map(&:to_sym)
+
+    (legacy_params & given_params).any?
+  end
+
+  def redirect_legacy_params
```

Comment:
Naming suggestion - `redirect_transforming_legacy_params`? Maybe just me, but when I read this method I assume it's returning me some params for a redirect, not actually doing the redirect itself.

Reply:
Also, do you think there's a sensible way of splitting this method up into slightly smaller ones? Find it slightly hard to parse this. Edit: Ah, another thing I should have waited to read the next commit for! Maybe you could move the refactor splitting out the taxons stuff into this commit?

Reply:
Thanks for reviewing @BeckaL.  I've moved all of the refactoring into the first commit, so the diff is clearer between the two.